src/components/Card/index.js

```diff
@@ -23,71 +23,98 @@ const Card = ({ className, item , data , index}) => {
     const [open, setOpen] = useState(false);
     const [isClickedBuyItem, setIsClickedBuyItem] = useState(false);
     const [openedMediaIndex, setOpenedMediaIndex] = useState(0);
+    const [price, setPrice] = useState("")
+    const [isMInted, setIsMinted] = useState(null)
 
     const scrollRef = useRef(null);
 
     const auth = useSelector(state => state.authReducer.data);
 
+    const _Contract = new web3.eth.Contract(
+        artTokenContractABI.abi,
+        item.collection_address
+    );
+
+    useEffect( async () => {
+        let weiPrice = await _Contract.methods.MINT_PRICE().call()
+        let number = Web3.utils.fromWei(weiPrice, 'ether');
+        setPrice(number);
+        setIsMinted(item.token_id);
+    }, []);
+
     useEffect(() => {
         open ? disableBodyScroll(scrollRef) : enableBodyScroll(scrollRef);
         setOpenedMediaIndex(index);
     }, [open]);
 
     const buyNFT = async (data) => {
-        if (data.owner !== auth.address) {
+        // if (data.owner !== auth.address) {
             setIsClickedBuyItem(true);
             try {
-                const tokenId = data.idForSale;
-                const price = data.price;
-                let toWei = web3.utils.toWei(price, 'ether');
-                await _Contract.methods
-                    .buyItem(tokenId)
-                    .send({ from: auth.address, value: toWei, gas: 250000 })
-                let res = await Actions.updateNFT({ tokenId: tokenId, buyer: auth.address });
-                if (res) {
-                    toast.success("Success!", toastOptions);
-                } else {
-                    toast.error("Some went wrong!", toastOptions);
-                }
+
+                _Contract.events.TokenMinted().on('data', (event) => {
+                    console.log(`token minted: ${event.returnValues._tokenId} , ${event.returnValues._metadataId}`);
+                    updateNFT(event.returnValues._tokenId, event.returnValues._metadataId);
+                })
```

Comment:
Why is this in frontend? This part should be in backend.
Backend listens smart contract events. Not frontend.